src/components/dictionaryConcepts/components/ViewMappingsModal.jsx

```diff
@@ -1,91 +1,115 @@
-                minWidth: 100,
-              },
-              {
-                Header: 'To Concept Name',
-                accessor: 'to_concept_name',
-              },
-              {
-                Header: 'Map Type',
-                accessor: 'map_type',
-              },
-              {
-                Header: 'Dictionary',
-                accessor: 'source',
-              },
-              {
-                Header: 'Action',
-                id: 'row',
-                filterable: false,
-                sortable: false,
-                Cell: row => (
-                  <AddMapping
-                    buttonName="Edit mapping"
-                    to_concept_url={mappings[row.index].to_concept_url}
-                    url={mappings[row.index].url}
-                    map_type={mappings[row.index].map_type}
-                    to_concept_name={mappings[row.index].to_concept_name}
-                    source={source}
-                    editMapping={editMapping}
-                    concepts={concepts}
-                  />
-                ),
-              },
-            ]}
-            className="-striped -highlight custom-table-width custom-table-min-height"
-          />
-        </div>
-      </ModalBody>
-      <ModalFooter>
-        <Button color="secondary" onClick={handleToggle}>
+export class ViewMappingsModal extends Component {
+  render() {
+    const {
+      handleToggle,
+      modal,
+      mappings,
+      mappingLimit,
+      displayName,
+      source,
+      editMapping,
+      concepts,
+    } = this.props;
+
+    return (
+      <div className="col-9">
+        <Modal isOpen={modal} className="modal-lg">
+          <ModalHeader>
+            {`${displayName} Mappings`}
+          </ModalHeader>
+          <ModalBody>
+            <div className="row col-12 custom-concept-list">
+              <ReactTable
+                data={mappings}
+                defaultPageSize={mappings.length <= mappingLimit ? mappings.length : mappingLimit}
+                filterable
+                noDataText="No mappings!"
+                minRows={0}
+                columns={[
+                  {
+                    Header: 'From Concept Name',
+                    accessor: 'from_concept_name',
+                    minWidth: 100,
+                  },
+                  {
+                    Header: 'To Concept Name',
+                    accessor: 'to_concept_name',
+                  },
+                  {
+                    Header: 'Map Type',
+                    accessor: 'map_type',
+                  },
+                  {
+                    Header: 'Dictionary',
+                    accessor: 'source',
+                  },
+                  {
+                    Header: 'Action',
+                    id: 'row',
+                    filterable: false,
+                    sortable: false,
+                    Cell: row => (
+                      <AddMapping
```

Comment:
@abulojoshua1 you can spread mapping here so that you don't have to use `row.index`. look at how I did it [Here](https://github.com/openmrs/openmrs-ocl-client/pull/226/files#diff-cee9e0a0dfc4f21601c8a51173062484).

Reply:
@dessHub, I cannot seem to find the actual reference you mean in the link, could you be more specific and provide a reference to a file?

Reply:
Thanks for the help @dessHub.